src/pages/bot-firewall/index.tsx

```diff
@@ -121,68 +136,116 @@ export const BotFirewall: NextPage<CustomPageProps> = ({ embed }) => {
 
   return (
     <>
-      <UseCaseWrapper useCase={USE_CASES.botFirewall} embed={embed} contentSx={{ maxWidth: 'none' }}>
+      <UseCaseWrapper
+        useCase={USE_CASES.botFirewall}
+        embed={embed}
+        contentSx={{ maxWidth: 'none' }}
+        instructionsNote={`For the purposes of this demo, you can only block/unblock your own IP address (${visitorData?.ip}). The block expires after one hour. The database of bot visits is cleared on every website update.`}
+      >
         <div className={styles.container}>
-          <h2 className={styles.title}>Detected bot visits</h2>
-          <Button
-            size="small"
-            outlined
-            onClick={() => {
-              refetchBotVisits();
-              refetchBlockedIps();
-            }}
-            className={styles.reloadButton}
-            disabled={isLoading}
-          >
-            {isLoading ? 'Loading ⏳' : 'Reload'}
-          </Button>
-          <i>
-            Note: For the purposes of this demo, you can only block/unblock your own IP address ({visitorData?.ip}). The
-            block expires after one hour. The database of bot visits is cleared on every website update.
-          </i>
-          <table className={styles.ipsTable}>
+          <div className={styles.header}>
+            <h2 className={styles.title}>Detected bot visits</h2>
+            <Button
+              size="small"
+              outlined
+              onClick={() => {
+                refetchBotVisits();
+                refetchBlockedIps();
+              }}
+              className={styles.reloadButton}
+              disabled={isLoading}
+            >
+              {isLoading ? 'Loading ⏳' : 'Reload bot visits'}
```

Comment:
Great catch! 

* First problem was that the loading state was not showing at all because I was incorrectly using `isLoading` instead of `isFetching` from react-query. 
* Second problem is that the requests are still really fast, around 100-150ms. Even if you add a "Reload finished" state, you get an ugly UI flash of "Loading" that is too quick to read, so it's still unpleasant/jarring. 
  * I considered just removing the loading state, on the other hand the request might take longer on slower connections, it would be useful to have there. 
  * Instead, I just added an artificial wait time of 500ms to the button. Fast enough to not be annoying, slow enough to read and recognize that the button is definitely doing something and you have fresh data when it's done. 
  
  What do you think? https://github.com/fingerprintjs/fingerprintjs-pro-use-cases/pull/112/commits/f3e7321f1853f963db1c2756830dfeb86158ee49

Reply:
Looks good! 😁